File: 713-subarray-product-less-than-k/subarray-product-less-than-k.py
class Solution(object):
    def numSubarrayProductLessThanK(self, nums, k):
        if k<=1:
            return 0
        l=0
        product=1
        count=0
        for i in range(len(nums)):
            product*=nums[i]
            while(product>=k):
                product=product//nums[l]
                l+=1
            count+=(i-l+1)
        return count

File: 713-subarray-product-less-than-k/subarray-product-less-than-k.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
       int l=0;
       int pro=1;
       int c=0;
       if(k<=1) return 0;
       for(int i=0;i<nums.length;i++){
        pro*=nums[i];
        while(pro>=k){
            pro=pro/nums[l];
            l++;
        }
        c+=(i-l+1);
       } 
    return c;
    }
}